moto/clouddirectory/responses.py

```diff
@@ -0,0 +1,93 @@
+"""Handles incoming clouddirectory requests, invokes methods, returns responses."""
+
+import json
+
+from moto.core.responses import BaseResponse
+
+from .models import CloudDirectoryBackend, clouddirectory_backends
+
+
+class CloudDirectoryResponse(BaseResponse):
+    """Handler for CloudDirectory requests and responses."""
+
+    def __init__(self) -> None:
+        super().__init__(service_name="clouddirectory")
+
+    @property
+    def clouddirectory_backend(self) -> "CloudDirectoryBackend":
+        """Return backend instance specific for this region."""
+        return clouddirectory_backends[self.current_account][self.region]
+
+    def create_directory(self) -> str:
+        name = self._get_param("Name")
+        schema_arn = self._get_param("SchemaArn")
+        directory = self.clouddirectory_backend.create_directory(
+            name=name,
+            schema_arn=schema_arn,
+        )
+
+        return json.dumps(
+            dict(
+                DirectoryArn=directory.directory_arn,
+                Name=name,
+                ObjectIdentifier=directory.object_identifier,
+                AppliedSchemaArn=directory.schema_arn,
+            )
+        )
+
+    def list_directories(self) -> str:
+        next_token = self._get_param("NextToken")
+        # max_results = self._get_param("MaxResults")
+        state = self._get_param("State")
+        directories = self.clouddirectory_backend.list_directories(
+            state=state,
+        )
+        directory_list = [directory.to_dict() for directory in directories]
+        return json.dumps(dict(Directories=directory_list, NextToken=next_token))
```

Comment:
This should only return a `NextToken` if pagination is actually implemented - otherwise boto3 will end up in an endless loop when using `boto3_client.get_paginator('list_directories')`